packages/template-retail-react-app/app/hooks/use-product-pricing.js

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright (c) 2023, Salesforce, Inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+import {useMemo} from 'react'
+import {getDisplayPrice} from '../utils/product-utils'
+
+/**
+ * @typedef {object} ProductPricing
+ * @property {string} currency
+ * @property {string} basePrice
+ * @property {string} discountPrice
+ */
+
+/**
+ * Gets the product pricing, allowing for extensibility using custom hooks
+ * @param {object} product
+ * @param {number} quantity
+ * @return {ProductPricing}
+ */
+function useProductPricing(product, quantity = 1) {
+    let qty = Number(quantity)
+    if (qty < 1 || isNaN(qty) || !Number.isInteger(qty)) {
+        qty = 1
+    }
```

Comment:
🤔 Would it be better to throw if the input is invalid, rather than replacing it with a default? Also, a non-integer quantity still seems valid for products like bulk foods sold by weight. 🤷

Reply:
not in my opinion, no, it would throw errors in tiles, pdp, cart, etc.. I removed the `isInteger`